components/withBreadcrumbs.tsx

```diff
@@ -20,11 +21,55 @@ const WithBreadcrumbs: FC = () => {
       return [];
     }
 
-    return getSideNavigation([navigationKey as NavigationKeys])
-      .map(([, item]) => item.items)
-      .flat()
-      .filter(([, item]) => pathname.includes(item.link))
-      .map(([, item]) => ({ label: item.label, href: item.link }));
+    const navigationTree: Array<[string, MappedNavigationEntry]> =
+      getSideNavigation([navigationKey as NavigationKeys]);
+
+    const toCamelCase = (str: string) => {
+      return str
+        .split('-')
+        .map((word, index) =>
+          index === 0 ? word : word[0].toUpperCase() + word.slice(1)
+        )
+        .join('');
+    };
+
+    const pathList = pathname
+      .split('/')
+      .filter(item => {
+        return item !== '';
+      })
+      .map(path => {
+        return toCamelCase(path);
+      });
+
+    const findPathInTree = (
```

Comment:
This seems super confusing and the time complexity of this is also not great. I'm quite sure you can solve this issue by using `.flat()` (to flatten the tree, as we don't need to navigate on the actual tree) and use simple ES6 builtins.

@canerakdas if you have time can you give a look into this? The old code was working at some point so I assume the current issue must be something silly